File: sherpa-onnx/java-api/src/main/java/com/k2fsa/sherpa/onnx/OnlineToneCtcModelConfig.java
package com.k2fsa.sherpa.onnx;

public class OnlineToneCtcModelConfig {
    private final String model;

    private OnlineToneCtcModelConfig(Builder builder) {
        this.model = builder.model;
    }

    public static Builder builder() {
        return new Builder();
    }

    public String getModel() {
        return model;
    }

    public static class Builder {
        private String model = "";

        public OnlineToneCtcModelConfig build() {
            return new OnlineToneCtcModelConfig(this);
        }

        public Builder setModel(String model) {
            this.model = model;
            return this;
        }
    }
}

File: sherpa-onnx/csrc/online-t-one-ctc-model-config.cc
// sherpa-onnx/csrc/online-t-one-ctc-model-config.cc
//
// Copyright (c)  2025  Xiaomi Corporation

#include "sherpa-onnx/csrc/online-t-one-ctc-model-config.h"

#include "sherpa-onnx/csrc/file-utils.h"
#include "sherpa-onnx/csrc/macros.h"

namespace sherpa_onnx {

void OnlineToneCtcModelConfig::Register(ParseOptions *po) {
  po->Register("t-one-ctc-model", &model,
               "Path to CTC model.onnx from T-one. Please see "
               "https://github.com/k2-fsa/sherpa-onnx/pull/2571");
}

bool OnlineToneCtcModelConfig::Validate() const {
  if (!FileExists(model)) {
    SHERPA_ONNX_LOGE("T-one CTC model '%s' does not exist", model.c_str());
    return false;
  }

  return true;
}

std::string OnlineToneCtcModelConfig::ToString() const {
  std::ostringstream os;

  os << "OnlineToneCtcModelConfig(";
  os << "model=\"" << model << "\")";

  return os.str();
}

}  // namespace sherpa_onnx
